core/src/main/java/com/linecorp/armeria/server/ServiceRequestContext.java

```diff
@@ -48,9 +52,39 @@
     PathMapping pathMapping();
 
     /**
+     * Returns the path parameters mapped by the {@link PathMapping} associated with the {@link Service}
+     * that is handling the current {@link Request}.
+     */
+    Map<String, String> pathParams();
+
+    /**
+     * Returns the value of the specified path parameter.
+     */
+    default String pathParam(String name) {
+        return pathParams().get(name);
+    }
+
+    /**
+     * Returns the path with its prefix removed. This method can be useful for a reusable service bound
+     * at various path prefixes.
+     *
+     * @return the path with its prefix removed, starting with '/'
+     *
+     * @throws IllegalStateException if the {@link PathMapping} associated with this request does not have
+     *                               a prefix
+     */
+    default String pathWithoutPrefix() {
```

Comment:
It would have to be on DefaultRequestContext but just the normal
```
private String pathWithoutPrefix;

String pathWithoutPrefix() {
  if (pathWithoutPrefix != null) {
    return pathWithoutPrefix;
  }
  pathWithoutPrefix = ...substring...
  return pathWithoutPrefix;
}
```

Not a big deal, but this method smells like it shouldn't have logic when calling it multiple times.